CI job for generating spack package is failing.

Log output:
`##[error]Named volume "amd64-ubuntu-20.04-gcc-12-gcc-10-cache:/build:delegated" is used in service "ubuntu-cpp" but no declaration was found in the volumes section.
##[error]The process '/usr/local/bin/docker-compose' failed with exit code 1
`

--- a/ci/azure/azure-gcc-10-ubuntu-openmpi-spack.yml
+++ b/ci/azure/azure-gcc-10-ubuntu-openmpi-spack.yml
@@ -19,6 +19,7 @@ variables:
   UBUNTU: 20.04
   COMPILER_TYPE: gnu
   COMPILER: gcc-10
+  HOST_COMPILER: gcc-10
   BUILD_TYPE: release
   ULIMIT_CORE: 0
   CODE_COVERAGE: 0

